core/slice.go

```diff
@@ -269,9 +276,11 @@ func (sl *Slice) Append(header *types.Header, domPendingHeader *types.Header, do
 		"uncles", len(block.Uncles()), "txs", len(block.Transactions()), "etxs", len(block.ExtTransactions()), "gas", block.GasUsed(),
 		"root", block.Root(),
 		"order", order,
-		"elapsed", common.PrettyDuration(time.Since(start)))
+		"elapsed", common.PrettyDuration(time.Since(start)),
+		"tps", tps,
+	)
 
-	return localPendingEtxs, nil
+	return localPendingEtxs, tps, nil
```

Comment:
Reporting the computed TPS after every append will result in erroneous measurements in two cases: syncing and reorging.

When you sync, you might download 1000s of blocks and submit thousands of TPS measurements. These TPS measurements will be historical and not representative of actual TPS now. These measurements should not be reported.

When you reorg, you effectively "un-process" transactions. Currently you have no logic to remove TPS entries for rolled-back blocks. This will result in erroneous measurements.